cmd/fluxctl/install_cmd.go

```diff
@@ -25,49 +29,87 @@ func (opts *installOpts) Command() *cobra.Command {
+	cmd.Flags().StringVar(&opts.params.GitURL, "git-url", "",
 		"URL of the Git repository to be used by Flux, e.g. git@github.com:<your username>/flux-get-started")
-	cmd.Flags().StringVar(&opts.GitBranch, "git-branch", "master",
+	cmd.Flags().StringVar(&opts.params.GitBranch, "git-branch", "master",
 		"Git branch to be used by Flux")
-	cmd.Flags().StringSliceVar(&opts.GitPaths, "git-paths", []string{},
-		"relative paths within the Git repo for Flux to locate Kubernetes manifests")
-	cmd.Flags().StringSliceVar(&opts.GitPaths, "git-path", []string{},
-		"relative paths within the Git repo for Flux to locate Kubernetes manifests")
-	cmd.Flags().StringVar(&opts.GitLabel, "git-label", "flux",
+	cmd.Flags().StringSliceVar(&opts.params.GitPaths, "git-path", []string{},
+		"Relative paths within the Git repo for Flux to locate Kubernetes manifests")
+	cmd.Flags().StringVar(&opts.params.GitLabel, "git-label", "flux",
 		"Git label to keep track of Flux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
-	cmd.Flags().StringVar(&opts.GitUser, "git-user", "Flux",
-		"username to use as git committer")
-	cmd.Flags().StringVar(&opts.GitEmail, "git-email", "",
-		"email to use as git committer")
-	cmd.Flags().BoolVar(&opts.GitReadOnly, "git-readonly", false,
-		"tell flux it has readonly access to the repo")
-	cmd.Flags().BoolVar(&opts.ManifestGeneration, "manifest-generation", false,
-		"whether to enable manifest generation")
-	cmd.Flags().StringVar(&opts.Namespace, "namespace", "",
-		"cluster namespace where to install flux")
-	cmd.Flags().BoolVar(&opts.RegistryScanning, "registry-scanning", true,
-		"scan container image registries to fill in the registry cache")
-	cmd.Flags().StringVarP(&opts.outputDir, "output-dir", "o", "", "a directory in which to write individual manifests, rather than printing to stdout")
+	cmd.Flags().StringVar(&opts.params.GitUser, "git-user", "Flux", "Username to use as git committer")
+	cmd.Flags().StringVar(&opts.params.GitEmail, "git-email", "", "Email to use as git committer")
+	cmd.Flags().BoolVar(&opts.params.ConfigAsConfigMap, "config-as-configmap", false,
+		"Create a ConfigMap to hold the Flux configuration given with --config-file. If false, a Secret will be used. Ignored if --config-file is not given.")
+	cmd.Flags().BoolVar(&opts.params.GitReadOnly, "git-readonly", false, "Tell flux it has readonly access to the repo")
+	cmd.Flags().BoolVar(&opts.params.ManifestGeneration, "manifest-generation", false, "Whether to enable manifest generation")
+	cmd.Flags().StringVar(&opts.params.Namespace, "namespace", "", "Cluster namespace in which to install Flux")
+	cmd.Flags().BoolVar(&opts.params.RegistryScanning, "registry-scanning", true, "Scan container image registries to fill in the registry cache")
+
+	// These are flags for control of the output, etc.
+	cmd.Flags().StringVar(&opts.configFile, "config-file", "", "Make this file into a secret or configmap for Flux to mount as config")
```

Comment:
```suggestion
	cmd.Flags().StringVar(&opts.configFile, "config-file", "", "Make this file into a secret (or configmap if --config-as-configmap is supplied) for Flux to mount as config")
```

Reply:
We should also explain what formats are supported for the config file. I think we should be restrictive the format (toml or yaml, for instance) instead of allowing every format viper supports.

(The minimum requirement IMO is that it can support key-value pairs and multiple git repos)

Reply:
My reasoning to restricting it to a single format is that:
1. Right now we only need key-values but we may want to store something more complicated than that in the future (e.g. multiple git repos) and not all formats will support them in the same way.
2. I think it can make it confusing for the end user (and will also make support more complicated) if we support every format viper supports.

Reply:
Related thing: I'm tempted to make all the field names `camelCase`, rather than using hyphens, since the latter look pretty strange in a YAML file. Version 2 of the format could even group the arguments into substructs -- though at that point, I would start to worry about it being confused for the Helm chart values (which do that too, in a reasonably thought-through but different way).

Reply:
> I'm tempted to make all the field names `camelCase`

It is not .. easy .. to have the flags as `--git-user` and the config file entries as `gitUser`, understandably but annoyingly. Specifically: `viper.BindPFlags` will use all the hypehated flag names, while viper.Decode will use all the camelCase field names, and they won't be related to the flag names. Ugh.